Makefile.dep

```diff
@@ -786,6 +786,10 @@ ifneq (,$(filter random,$(USEMODULE)))
     USEMODULE += hashes
   endif
 
+  ifneq (,$(filter prng_hwrng,$(USEMODULE)))
```

Comment:
Is it really a PRNG if it is generating random numbers from hardware? ;-)

Reply:
Yea that feels like a hack - we could rename them all `prng_%` -> `rng_impl_%`  

Having a single API for randomness in `random.h` makes sense - the caller doesn't care if that randomness is provided by a PRNG or a HWRNG

Reply:
They might, when it comes to timing, or when the deterministic behavior of a PRNG is actually required for some reason.

Reply:
> They might, when it comes to timing

In most cases the HWRNG will be plenty fast. It could be faster (#11924 :wink:), but hardly noticeable.

>  or when the deterministic behavior of a PRNG is actually required

Then the application should use a private instance of a PRNG.
The system RNG will be called by 'everyone', thereby severely hampering determinism.

Also mind you, this is just an option. An application developer can chose to select the HWRNG for everything and will get

 - cryptographically sound randomness
 - reduced code size 

if they accept the drawbacks you mentioned.

Reply:
> if they accept the drawbacks you mentioned.

Perfect call-in for a documentation upgrade ;-).

Reply:
I would add it to the documentation about the characteristics of the different PRNG implementations, but I couldn't find those either.

Reply:
For most PRNG you can look it up in literature just searching for the name. This is not the case for the HWNRG.